arangod/Aql/AqlValue.cpp

```diff
@@ -552,6 +553,50 @@ AqlValue AqlValue::get(CollectionNameResolver const& resolver,
   return AqlValue(AqlValueHintNull());
 }
 
+/// @brief get the (object) element by name
+AqlValue AqlValue::get(CollectionNameResolver const& resolver,
+                       arangodb::velocypack::StringRef const& name, bool& mustDestroy, bool doCopy) const {
+  mustDestroy = false;
+  switch (type()) {
+    case VPACK_SLICE_POINTER:
+      doCopy = false;
+    // intentionally falls through
+    case VPACK_INLINE:
+    // intentionally falls through
+    case VPACK_MANAGED_SLICE:
+    // intentionally falls through
+    case VPACK_MANAGED_BUFFER: {
+      VPackSlice s(slice());
+      if (s.isObject()) {
+        VPackSlice found(s.get(name));
+        if (found.isCustom()) {
```

Comment:
do we actually validate that the only custom field is `_id` ?

Reply:
At the moment, we don't validate this. But all our code uses the custom field type just for `_id`.